mangaki/mangaki/models.py

```diff
@@ -89,6 +89,20 @@ def random(self):
             nb_dislikes__lte=RANDOM_MAX_DISLIKES,
             nb_likes__gte=F('nb_dislikes') * RANDOM_RATIO)
 
+    def group_by_category(self):
+        """
+        Groups this queryset by category.
+
+        Returns:
+          by_category -- A mapping from category IDs to the list of works in
+              this queryset in the corresponding category. Order inside a
+              category is preserved.
+        """
+        by_category = {}
```

Comment:
Not sure how to do it using Django's ORM since I want this to preserve order inside a category group. Besides, you should only be using this method (like e.g. `in_bulk`) when you actually want the items.